gossip/src/contact_info.rs

```diff
@@ -366,6 +366,29 @@ impl ContactInfo {
         LegacyContactInfo::is_valid_address(addr, socket_addr_space)
     }
 
+    /// New random ContactInfo for tests and simulations.
+    pub fn new_rand<R: rand::Rng>(rng: &mut R, pubkey: Option<Pubkey>) -> Self {
+        let delay = 10 * 60 * 1000; // 10 minutes
+        let now = solana_sdk::timing::timestamp() - delay + rng.gen_range(0..2 * delay);
+        let pubkey = pubkey.unwrap_or_else(solana_sdk::pubkey::new_rand);
+        let mut node = ContactInfo::new_localhost(&pubkey, now);
+        let _ = node.set_gossip((Ipv4Addr::LOCALHOST, rng.gen_range(1024..u16::MAX)));
+        node
+    }
+
+    /// Construct a ContactInfo that's only usable for gossip
+    pub fn new_gossip_entry_point(gossip_addr: &SocketAddr) -> Self {
+        let mut node = Self::new(
+            Pubkey::default(),
+            solana_sdk::timing::timestamp(), // wallclock
+            0,                               // shred_version
+        );
+        if let Err(err) = node.set_gossip(*gossip_addr) {
+            error!("Invalid entrypoint: {gossip_addr}, {err:?}");
+        }
+        node
+    }
```

Comment:
equivalent to:
https://github.com/anza-xyz/agave/blob/fb3bf5534/gossip/src/legacy_contact_info.rs#L167-L175